src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -707,253 +1218,90 @@ namespace nbl::ext::imgui
 				.minDepth = 0.0f,
 				.maxDepth = 1.0f,
 			};
-			commandBuffer->setViewport(0, 1, &viewport);
 
-			// Setup scale and translation:
-			// Our visible imgui space lies from draw_data->DisplayPps (top left) to draw_data->DisplayPos+data_data->DisplaySize (bottom right). DisplayPos is (0,0) for single viewport apps.
+			commandBuffer->setViewport(0, 1, &viewport);
 			{
-				PushConstants constants{};
-				constants.scale[0] = 2.0f / drawData->DisplaySize.x;
-				constants.scale[1] = 2.0f / drawData->DisplaySize.y;
-				constants.translate[0] = -1.0f - drawData->DisplayPos.x * constants.scale[0];
-				constants.translate[1] = -1.0f - drawData->DisplayPos.y * constants.scale[1];
-
-				commandBuffer->pushConstants(pipeline->getLayout(), IShader::E_SHADER_STAGE::ESS_VERTEX, 0, sizeof(constants), &constants);
+				if (scissors.empty())
+				{
+					VkRect2D defaultScisors[] = { {.offset = {(int32_t)viewport.x, (int32_t)viewport.y}, .extent = {(uint32_t)viewport.width, (uint32_t)viewport.height}} };
+					commandBuffer->setScissor(defaultScisors); // cover whole viewport (dynamic scissors must be set only to not throw validation errors)
+				}
+				else
+					commandBuffer->setScissor(scissors);
 			}
+			
+			/*
+				Setup scale and translation, our visible imgui space lies from draw_data->DisplayPps (top left) to 
+				draw_data->DisplayPos+data_data->DisplaySize (bottom right). DisplayPos is (0,0) for single viewport apps.
+			*/
 
-			// Will project scissor/clipping rectangles into frame-buffer space
-			ImVec2 const clip_off = drawData->DisplayPos;         // (0,0) unless using multi-viewports
-			ImVec2 const clip_scale = drawData->FramebufferScale; // (1,1) unless using retina display which are often (2,2)
-
-			// Render command lists
-			// (Because we merged all buffers into a single one, we maintain our own offset into them)
-			int global_vtx_offset = 0;
-			int global_idx_offset = 0;
-			for (int n = 0; n < drawData->CmdListsCount; n++)
 			{
-				const ImDrawList* cmd_list = drawData->CmdLists[n];
-				for (int cmd_i = 0; cmd_i < cmd_list->CmdBuffer.Size; cmd_i++)
+				PushConstants constants
 				{
-					const ImDrawCmd* pcmd = &cmd_list->CmdBuffer[cmd_i];
+					.elementBDA = { mdiBuffer->getDeviceAddress() + mdiOffsets[MDI::EBC_ELEMENT_STRUCTURES]},
+					.elementCount = { mdiParams.drawCount },
```

Comment:
what do you need to know the draw count within the current MDI for on the shader side?

Reply:
old, unsued + to be nuked